### ISSUE TITLE
Enhancement: Integration with The-Federation.info for Additional Spam Detection Metrics

### ISSUE THREAD

**Issue by AlphaPro06** [db0/lemmy-overseer#2]
**Is your feature request related to a problem? Please describe.**
Currently, this service uses the Lemmy Fediverse Observer to identify and counter bad actors. While this provides valuable insights, I believe the integration with additional data sources could further improve spam detection.

**Describe the solution you'd like**
I suggest integrating with [The-Federation.info](https://the-federation.info/platform/73). This service offers real-time data, including user count, active users, posts, and comments, for various federated social networks. Incorporating this data could enhance our understanding of the instance's activity and improve spam detection capabilities.

**Describe alternatives you've considered**
Sticking with the current approach solely based on the Lemmy Fediverse Observer.

**Additional context**
The integration with The-Federation.info could potentially provide another layer of information to detect suspicious activity quickly and more accurately.

**Comment by AlphaPro06** [comment]
There is this spreadsheet I found as well that could useful and related to this issue

https://docs.google.com/spreadsheets/d/e/2PACX-1vRthB7RtY4Rr0t5fhVKaliJnwSmptMc5oJi7uha_OBcF4wpu4eElxAxNzaCqjlq6NsOE9GpgSnMzZ2x/pubhtml

**Comment by AlphaPro06** [comment]
I don't think it has a REST API but you use the following url using GraphiQL: https://the-federation.info/v1/graphql 

That is where I think they're using to get the data. More info on that `[here](https://the-federation.info/info)`

**Comment by db0** [comment]
That graphql link you sent me doesn't seem to work. You mean this? https://the-federation.info/graphql

If so, I have no idea how to use it as it doesn't seem to have any guidance

**Comment by morcantium** [comment]
You can use the link https://the-federation.info/graphql as a GUI version to write and run some test queries. For an example simple query format:

`query { nodes { id, host }}`

Will generate:

![Screenshot 2023-06-23 185427](https://github.com/db0/fediseer/assets/120954705/84430848-3bc7-40cc-94b8-901e464961d1)

The schema is helpfully included to the right of the GUI in screen shot above so you can see all the available fields for your queries.

Once you have built your graphql query you can just CURL or fetch the API endpoint at https://the-federation.info/v1/graphql with your JSON encoded query. There's some simple examples here: https://www.apollographql.com/blog/graphql/examples/4-simple-ways-to-call-a-graphql-api/ of how to achieve this.

I'm not associated in any way with the-federation.info by the way, just thought it may be of some small help in understanding their API setup if you think it would be useful to pull data from.
